src/read.rs

```diff
@@ -941,7 +955,7 @@ fn central_header_to_zip_file_inner<R: Read>(
         extra_field: Some(Arc::new(extra_field)),
         central_extra_field: None,
         file_comment,
-        header_start: offset,
+        header_start: offset.into(),
```

Comment:
Can header_start be changed to a `u32` so that this `into()` isn't necessary?

Reply:
GREAT comment! I will figure this out :)

Reply:
So it turns out this is used for both zip32 and zip64 results, but I had some difficulty understanding what was happening in `get_metadata()` (it is a complex function, so intrinsically difficult). I think this is actually an easy fix (make `offset` a `u64`), but I'm probably going to try to refactor `get_metadata()`, because while I am pretty sure the current code is correct, I want to clarify the control flow to make sure. I believe this will be an isolated change and easy to review (so worth including in this PR), and the result should be easier to read (or I will simply drop it and just make `offset` a `u64`).